portality/migrate/reapplication/reapp_init.py

```diff
@@ -2,97 +2,118 @@
 from datetime import datetime as dt
 from portality import reapplication
 from portality.core import app
-import os
+import os, time
 import codecs
 import csv
 from portality.clcsv import UnicodeWriter
 
+class ReappCsvException(Exception):
+    def __init__(self, message, account_id):
+        super(ReappCsvException, self).__init__(message)
+        self.account_id = account_id
 
-def delete_existing_reapp():
+def delete_existing_reapps():
     """Note that this cannot be run in isolation, because it leaves behind unresolved current_application fields in journals"""
 
-    # delete all the suggestions
+    # delete all the suggestions in a reapplication status
     q = models.SuggestionQuery(statuses=['reapplication', 'submitted']).query()
     models.Suggestion.delete_by_query(query=q)
+    #models.Suggestion.refresh()
 
     # remove all of the current reapplication records
     models.BulkReApplication.delete_all()
+    # models.BulkReApplication.refresh()
 
-def create_reapplication(journal):
-    if not isinstance(journal, models.Journal):
-        journal = models.Journal.pull(journal)
-    if not isinstance(journal, models.Journal):
-        print "Unable to resolve Journal from identifier"
+    # FIXME: should probably remove all the old csvs too
```

Comment:
It opens csv-s in overwrite mode. But maybe to avoid confusion we should remove them. However we can't remove all of them if only one journal or account is being deleted - otherwise it'd defeat the purpose of having these options on this script as helpers during firefighting after reapplications go live. So really the way it's now is probably best - only touch (and overwrite) files when they're being generated.